### ISSUE TITLE
Labels field in (unweighted) histogram ggplot object incorrectly includes 'weight' 

### ISSUE THREAD

**Issue by gmbecker** [tidyverse/ggplot2#2659]
Weight appears to be incorrectly (I think?) included in the `labels` element of a ggplot object when geom_histogram is used with no weight specified. This does not happen in the (largely) equivalent geom_dotplot call.

I would expect these to agree. In particular, given my basic understanding of what `labels` means, I would expect weight to be excluded in both cases unweighted cases.  I understand that this is detail about the internals of ggplot objects, and that they don't constitute an API, and thus I generally cannot expect consistency across time or function. That said, I thought you might want to know given that as maintainers you would have to account for this inconsistency in your own code. 

The issue does not appear to occur in the previous, CRAN-published version I have readily available.

```r
library(ggplot2)
dat = head(diamonds, 50)

## dotplot, no weight in labels when weight unspecified
dp = ggplot(dat, aes(price)) + geom_dotplot()
dp$labels # no weight

## histogram, weight included  erroneously (?) in labels
hst = ggplot(dat, aes(price)) + geom_histogram()
hst$labels #weight
```

I believe that the relevant difference is that `StatBindot` has `non_missing_aes("weight")` whereas `StatBin` has `default_aes = aes(y = stat(count), weight = 1)` as their respective ways of dealing with the unweighted case. This would have been introduced in #1921 if I'm correct about the cause. 

I'm not confident enough about the exact meaning of `non_missing_aes` to immediately make this a pull-request, however, as I can't find it explicitly documented anywhere.

**Comment by hadley** [comment]
It is weird that a default aesthetic should appear in the labels. I'm not sure we'll change anything, but we should take a look just to make sure.

**Comment by clauswilke** [comment]
I believe this is the same issue as #2406. When stats define default aesthetics those make it into the labels. The relevant code is here:
https://github.com/tidyverse/ggplot2/blob/86c41aca8829289305fd737707b2b9bfbd56ce70/R/plot-construction.r#L159-L162

And here are some of the affected stats:

- `stat_bin`:
https://github.com/tidyverse/ggplot2/blob/429991746ef69d7d66b1b33bb4c24e89b521907f/R/stat-bin.r#L148

- `stat_density`:
https://github.com/tidyverse/ggplot2/blob/429991746ef69d7d66b1b33bb4c24e89b521907f/R/stat-density.r#L65

- `stat_density_2d`:
https://github.com/tidyverse/ggplot2/blob/429991746ef69d7d66b1b33bb4c24e89b521907f/R/stat-density-2d.r#L47

There are probably others.

**Comment by clauswilke** [comment]
Closed via #2406.
